File: back-end/models/Manufacturer.go
package models

import (
	"time"

	"github.com/go-playground/validator/v10"
)

type Manufacturer struct {
	IdManufacturer   uint      `gorm:"primaryKey" json:"id_manufacturer"`
	NameManufacturer string    `json:"name_manufacturer"`
	YearFounded      int       `json:"year_founded"`
	IsActive         bool      `json:"is_active"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

func (manufacturer *Manufacturer) Validate() error {
	validate := validator.New()
	return validate.Struct(manufacturer)
}

File: front-end/web/src/app/models/manufacturer.ts
export class Manufacturer {
  id_manufacturer: number;
  name_manufacturer: string;
  year_founded: number;
  is_active: boolean;

  constructor() {
    this.id_manufacturer = 0;
    this.name_manufacturer = '';
    this.year_founded = 0;
    this.is_active = true;
  }
}
